### ISSUE TITLE
Add torchtyping as dependency?

### ISSUE THREAD

**Issue by gavinmdouglas** [JSchlensok/VespaG#5]
Hi again,

Just a quick mention that I think `torchtyping` should be added as a dependency.

Before installing it (with pip), I was getting the below error at the point of generating predictions, but it was resolved after installing that package.

Cheers,

Gavin

```
│ /mfs/gdouglas/local/miniforge3/envs/vespag/lib/python3.10/site-packages/torch/nn/modules/linear. │
│ py:114 in forward                                                                                │
│                                                                                                  │
│   111 │   │   │   init.uniform_(self.bias, -bound, bound)                                        │
│   112 │                                                                                          │
│   113 │   def forward(self, input: Tensor) -> Tensor:                                            │
│ ❱ 114 │   │   return F.linear(input, self.weight, self.bias)                                     │
│   115 │                                                                                          │
│   116 │   def extra_repr(self) -> str:                                                           │
│   117 │   │   return 'in_features={}, out_features={}, bias={}'.format(                          │
│                                                                                                  │
│ ╭──────────────────────────────────── locals ────────────────────────────────────╮               │
│ │ input = tensor([[ 0.0795, -0.1561,  0.1679,  ...,  0.1603, -0.2746,  0.0323],  │               │
│ │         │   │   [ 0.0088, -0.0908,  0.0464,  ...,  0.0891, -0.1113, -0.0046],  │               │
│ │         │   │   [ 0.1057,  0.0064, -0.0143,  ...,  0.1393, -0.2387, -0.0604],  │               │
│ │         │   │   ...,                                                           │               │
│ │         │   │   [-0.1762,  0.1132, -0.0450,  ...,  0.0286,  0.2649,  0.2094],  │               │
│ │         │   │   [ 0.0842,  0.0347, -0.0799,  ..., -0.1230, -0.1176, -0.0504],  │               │
│ │         │   │   [ 0.1492, -0.0191, -0.0351,  ...,  0.0520, -0.0500, -0.0366]], │               │
│ │         │      dtype=torch.float64)                                            │               │
│ │  self = Linear(in_features=2560, out_features=256, bias=True)                  │               │
│ ╰────────────────────────────────────────────────────────────────────────────────╯               │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
RuntimeError: mat1 and mat2 must have the same dtype
```

**Comment by JSchlensok** [comment]
Hi,
`torchtyping` was dropped as a dependency and is not used anymore, and I can't reproduce the error from a fresh environment. Could you provide me with a full stack trace for the exception and the output of `conda list` and `pip list`?

**Comment by gavinmdouglas** [comment]
Sure, I actually just noticed that this error only occurs when the `--embeddings` is left out (i.e., only when the embeddings are generated, rather than read in from a file).

I re-ran the command in a new conda environment with and without this option:

```
python -m vespag predict \
	--input /mfs/gdouglas/projects/vespag/phydms_seqs/focal_seqs_w_10_pos.faa \
	--output /mfs/gdouglas/projects/vespag/phydms_seqs/vespag_out_TEST \
	--embeddings /mfs/gdouglas/projects/vespag/phydms_seqs/esm2_embeddings.h5 \
	> predict_w_existing_embed.stdout.txt \
	2> predict_w_existing_embed.stderr.txt

python -m vespag predict \
	--input /mfs/gdouglas/projects/vespag/phydms_seqs/focal_seqs_w_10_pos.faa \
	--output /mfs/gdouglas/projects/vespag/phydms_seqs/vespag_out_TEST2 \
	> predict_new_embed.stdout.txt \
	2> predict_new_embed.stderr.txt
```

You can see the resulting stdout and stderr files below (where `predict_new_embed.stderr.txt` is the one with the error).

* [predict_w_existing_embed.stdout.txt](https://github.com/JSchlensok/VespaG/files/15101364/predict_w_existing_embed.stdout.txt)
* [predict_w_existing_embed.stderr.txt](https://github.com/JSchlensok/VespaG/files/15101365/predict_w_existing_embed.stderr.txt)
* [predict_new_embed.stdout.txt](https://github.com/JSchlensok/VespaG/files/15101366/predict_new_embed.stdout.txt)
* [predict_new_embed.stderr.txt](https://github.com/JSchlensok/VespaG/files/15101361/predict_new_embed.stderr.txt)

Here is the `conda list` and `pip freeze` output:
* [conda_list.txt](https://github.com/JSchlensok/VespaG/files/15101362/conda_list.txt)
* [pip_freeze.txt](https://github.com/JSchlensok/VespaG/files/15101363/pip_freeze.txt)

**Comment by JSchlensok** [comment]
Thanks for the effort! I've pinpointed the issue to CPU embedding generation mistakenly using 64-bit instead of 32-bit precision and fixed it in 8980abd34ad9d5e77e12b6ee28e289b45ebe528e.

As for how installing `torchtyping` solves this, I'm still a bit baffled and can't get a good intuition from the package lists, so I'd close the issue for the time being unless the issue persists on your end.

**Comment by gavinmdouglas** [comment]
No worries! Glad to hear you could track down the fix.
